ast/Trees.h

```diff
@@ -954,7 +954,9 @@ EXPRESSION(Cast) {
     core::TypePtr type;
     ExpressionPtr arg;
 
-    Cast(core::LocOffsets loc, core::TypePtr ty, ExpressionPtr arg, core::NameRef cast);
+    ExpressionPtr typeExpr;
```

Comment:
This means that we have a `Loc` for the `typeExpr` now. Does that help make any error messages/autocorrects better now?

Reply:
I don't see anything obvious in handling `Cast` in `environment.cc` (assuming we propagated the extra information into `cfg::Cast` nodes), but maybe there are other places.